src/services/completions.ts

```diff
@@ -323,15 +323,25 @@ namespace ts.Completions {
             return result;
         }
 
+        /**
+         * Given a path ending at a directory, gets the completions for the path, and filters for those entries containing the basename.
+         */
         function getCompletionEntriesForDirectoryFragment(fragment: string, scriptPath: string, extensions: string[], includeExtensions: boolean, span: TextSpan, exclude?: string, result: CompletionEntry[] = []): CompletionEntry[] {
+            if (fragment === undefined) { fragment = "./"; }
+
+            fragment = normalizeSlashes(fragment);
+
+            const baseName = getBaseFileName(fragment);
+
+            // Remove the basename from our directory path
             fragment = getDirectoryPath(fragment);
-            if (!fragment) {
-                fragment = "./";
-            }
-            else {
-                fragment = ensureTrailingDirectorySeparator(fragment);
+
+            if (fragment === "") {
+                fragment = "." + directorySeparator;

```

Comment:
No, the comment was specifically about the fact that line 330 uses `"./"` while line 340 uses `"." + directorySeparator"`, though they do the same thing. Perhaps you should initialize `fragment` to `""` on 330 rather than repeat yourself on 340.